### PR TITLE
Implemented chatManager and ActivePassive chatCell logic implemented

### PR DESCRIPTION
**Pull request by hasanmainul**

### CODE REVIEW

File: LocalChat/LocalChat/ChatManager.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    
+    var chatInfo = [[String: AnyObject]]()
+  
+    override init() {
+        super.init()
+    }
+    
+//    func getChatInfo(completionHandler: (messageInfo: [[String: AnyObject]]) -> Void) {
+//        completionHandler(messageInfo: chatInfo)
+//    }
```
Comment (khalidmahmud) [inline LocalChat/LocalChat/ChatManager.swift:22]: if not needed remove the commented function

File: LocalChat/ChatViewController.swift [hunk 2]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+        return self.chatMessages.count;
     }
     
     func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        if indexPath.row % 2 == 0 {
+        var dict = chatMessages[indexPath.row]
+        let userName = dict["userName"] as? String
+        if userName != userSegmentControl.titleForSegmentAtIndex(userSegmentControl.selectedSegmentIndex) as String! {
             let cell = tableView.dequeueReusableCellWithIdentifier("passiveChatCell", forIndexPath: indexPath) as! PassiveChatCell
-            print(chatText)
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Please remove print

File: LocalChat/ChatViewController.swift [hunk 3]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
-            cell.passiveChatLabel?.text = chatText
+            cell.passiveChatLabel?.text = dict["message"] as? String
+            cell.passiveNameAndDateLabel?.text = dict["userName"] as? String
+            cell.backgroundColor = UIColor.clearColor()
             return cell
         } else {
             let cell = tableView.dequeueReusableCellWithIdentifier("activeChatCell", forIndexPath: indexPath) as! ActiveChatCell
-            print(chatText)
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift:99]: Please remove print

File: LocalChat/ChatViewController.swift [hunk 4]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
     
     func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 30;
+        return self.chatMessages.count;
     }
     
     func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        if indexPath.row % 2 == 0 {
+        var dict = chatMessages[indexPath.row]
+        let userName = dict["userName"] as? String
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 5]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+        if userName != userSegmentControl.titleForSegmentAtIndex(userSegmentControl.selectedSegmentIndex) as String! {
             let cell = tableView.dequeueReusableCellWithIdentifier("passiveChatCell", forIndexPath: indexPath) as! PassiveChatCell
-            print(chatText)
-            cell.passiveChatLabel?.text = chatText
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 6]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+            cell.passiveChatLabel?.text = dict["message"] as? String
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 7]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+            cell.passiveNameAndDateLabel?.text = dict["userName"] as? String
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 8]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+            cell.backgroundColor = UIColor.clearColor()
             return cell
         } else {
             let cell = tableView.dequeueReusableCellWithIdentifier("activeChatCell", forIndexPath: indexPath) as! ActiveChatCell
-            print(chatText)
-            cell.activeChatLabel?.text = chatText
+            cell.activeChatLabel?.text = dict["message"] as? String
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 9]
```diff
@@ -85,19 +89,23 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
...
+            cell.activeNameAndDateLabel?.text = dict["userName"] as? String
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Maintain a Constant file for key

File: LocalChat/LocalChat/ChatManager.swift [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+class ChatManager: NSObject {
+    static let sharedInstance = ChatManager()
+    
+    var chatInfo = [[String: AnyObject]]()
+  
+    override init() {
+        super.init()
+    }
+    
+//    func getChatInfo(completionHandler: (messageInfo: [[String: AnyObject]]) -> Void) {
```
Comment (jannatulabeda) [inline LocalChat/LocalChat/ChatManager.swift]: Remove unnecessary lines

File: LocalChat/LocalChat/ChatManager.swift (hunk 1 above)
Comment (jannatulabeda) [inline LocalChat/LocalChat/ChatManager.swift:22]: Remove unnecessary comments

File: LocalChat/LocalChat/ChatManager.swift [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+//        completionHandler(messageInfo: chatInfo)
+//    }
+    
+    func sendChatMessage(message: String, withUserName userName: String?) -> [[String: AnyObject]] {
+        var dict = [String: AnyObject]()
+        dict["message"] = message
```
Comment (jannatulabeda) [inline LocalChat/LocalChat/ChatManager.swift]: Maintain a Constant file for key

File: LocalChat/LocalChat/ChatManager.swift [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+        dict["userName"] = userName
```
Comment (jannatulabeda) [inline LocalChat/LocalChat/ChatManager.swift]: Maintain a Constant file for key

File: LocalChat/ChatViewController.swift [hunk 13]
```diff
@@ -18,6 +18,7 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
     var frameView: UIView!
     var numberOfSegment: Int = 2
```
Comment (jannatulabeda) [inline LocalChat/ChatViewController.swift]: Don't use hard coded value

File: LocalChat/LocalChat/ChatManager.swift (hunk 1 above)
Comment (hasanmainul) [inline LocalChat/LocalChat/ChatManager.swift:22]: Its needed for future reference, if not needed then will remove this

**Comment by khalidmahmud** [comment]
Ok

**Pull request merged**